sky/server/requests/process.py

```diff
@@ -67,14 +68,14 @@ def shutdown(self, wait: bool = True) -> None:
 
 
 # Define the worker function outside of the class to avoid pickling self
-def _disposable_worker(fn, initializer: Optional[Callable], initargs: Tuple,
-                       args, kwargs):
+def _disposable_worker(fn, initializer, initargs, result_queue, args, kwargs):
     try:
         if initializer is not None:
             initializer(*initargs)
-        fn(*args, **kwargs)
+        result = fn(*args, **kwargs)
+        result_queue.put(result)
```

Comment:
Do we use this at all right now? Or just the exception case?

Reply:
We only use the exception case, this change makes the interface of `DisposableExecutor` consistent with [`ProcessPoolExecutor`](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor) of std lib to avoid special handling for different executor impls. Kind of a code gardening